lib/gollum/app.rb

```diff
@@ -506,6 +506,19 @@ class App < Sinatra::Base
         end
       end
 
+      get %r{
+        /commit/  # match any URL beginning with /show/
+        (\w+)     # match the SHA1
+      }x do |version|
+        @version = version
+        wiki = wiki_new
+        @commit = wiki.repo.commit(version)
+        parent = @commit.parent
```

Comment:
Is it possible for `@commit` to be `nil` at this point? In that case, we need another `nil`-check, unfortunately

Reply:
What should gollum show to the users when some invalid sha1 is given?

Reply:
I think in that case a `mustache :error` would be appropriate, like [here](https://github.com/gollum/gollum/blob/master/lib/gollum/app.rb#L502-L503). Message could be something like `"Invalid commit: #{@version}"`.